### ISSUE TITLE
After upgraded to 0.10.1 i started to get Cannot read property 'state' of undefined error

### ISSUE THREAD

**Issue by hyurtseven81** [nuxt/nuxt.js#445]
Hi,

After i upgraded i started to get some errors.

`[nuxt.js] Cannot load components TypeError: Cannot read property 'state' of undefined
    at _callee$ (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:401:1), <anonymous>:27:22)
    at tryCatch (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:619:1), <anonymous>:64:40)
    at Generator.invoke [as _invoke] (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:619:1), <anonymous>:299:22)
    at Generator.prototype.(anonymous function) [as next] (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:619:1), <anonymous>:116:21)
    at step (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:401:1), <anonymous>:13:191)
    at eval (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:401:1), <anonymous>:13:437)
    at eval (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:401:1), <anonymous>:13:99)
    at eval (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:401:1), <anonymous>:51:17)
    at promisify (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:373:1), <anonymous>:99:15)
    at promiseSeries (eval at <anonymous> (http://localhost:3000/_nuxt/nuxt.bundle.a1d658b70038de92930d.js:373:1), <anonymous>:80:10)`

and on the terminal i get the following error:

`Error: Error: Request failed with status code 404
    at server-bundle.js:18129:13
    at process._tickDomainCallback (internal/process/next_tick.js:135:7)`

i use nuxt with express.js and on 0.9.9 version there is no problem.

Thank you.

**Comment by pi0** [comment]
@hyurtseven81 Please read [v0.10.1 release notes](https://github.com/nuxt/nuxt.js/releases/tag/v0.10.0) I think you didn't replace pages `data` with `asyncData` ;)

**Comment by evseevnn** [comment]
i think i was have same error today.
My [pull request](https://github.com/nuxt/nuxt.js/pull/442)

**Comment by hyurtseven81** [comment]
Actually i read and i changed them but there is still problem

**Comment by KonstantinVlasov** [comment]
Hi! Have the same issue with middleware.
**middleware/example.js**
```
export default function ({ store }) {
  console.log(store) // got undefined
}
```

**Comment by hyurtseven81** [comment]
By the way after i changed line 389 on client.js with this: "let context = getContext({ to: router.currentRoute, isClient: true, store })" the problem has solved, like @evseevnn 's pull request

**Comment by Atinux** [comment]
Sorry about this, the #442 has been merged and a new version [0.10.2](https://github.com/nuxt/nuxt.js/releases/tag/v0.10.2) has been released, please upgrade 🙏

**Comment by hyurtseven81** [comment]
Thank you for your quick reply and helps.
